app/services/registration_checker.rb

```diff
@@ -200,5 +216,26 @@ def existing_registration_in_series?
       end
       false
     end
+
+    def competitor_qualifies_for_event?(event, qualification)
+      competitor_qualification_results = UserApi.qualifications(@requestee_user_id)
+      result_type = qualification['resultType']
+
+      competitor_pr = competitor_qualification_results.find { |result| result['eventId'] == event && result['type'] == result_type }
+      return false if competitor_pr.blank?
+
+      return false unless Date.parse(competitor_pr['on_or_before']) <= Date.parse(qualification['whenDate'])
```

Comment:
Too much happening in this line. What if parsing fails?

Reply:
The latest commit declares them as variables before comparison, and I've also added error handling for the case where parsing fails - which I assume is what you wanted? 

It still fails "silently" in the sense that we just return false and it says that qualification isn't met, so a bit of debugging will be needed. At the same time, I don't see a clean way of returning a different error, and both date values come from our API so the input should be fairly consistent (as opposed to user-provided dates, which we'd of course need to parse more carefully)